packages/DynamicHyperlink/src/DynamicHyperlink.js

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import PropTypes from "prop-types";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import "./DynamicHyperlink.scss";
+
+export default function DynamicHyperlink({ onFetch }) {
+  const I18n = useI18n();
+
+  React.useEffect(() => {
+    function updateDynamicHyperlinks() {
+      const dynamicHyperlinks = Array.from(document.querySelectorAll("[data-dynamic-hyperlink]"));
+
+      dynamicHyperlinks
+        .filter(dynamicHyperlink => !dynamicHyperlink.hasAttribute("data-dynamic-hyperlink--processed"))
+        .forEach(dynamicHyperlink => {
+          const originalLinkUrl = dynamicHyperlink.innerHTML;
+          dynamicHyperlink.innerHTML = I18n.t("dynamicHyperlink.loading"); // eslint-disable-line no-param-reassign
+          dynamicHyperlink.setAttribute("data-dynamic-hyperlink--processed", "true");
+
+          // the consumer does the fetching of data
+          onFetch(originalLinkUrl).then(({ error, name, term }) => {
```

Comment:
onFetch should take in the value of the data-dynamic-hyperlink value.  Let's chat about the interface today

Reply:
onFetch(originalLinkUrl, LinkType)